qutebrowser/browser/greasemonkey.py

```diff
@@ -217,9 +216,11 @@ def _add_script(self, script):
         log.greasemonkey.debug("Loaded script: {}".format(script.name))
 
     def _required_url_to_file_path(self, url):
-        # TODO: Save to a more readable name
+        # Save to a more readable name
         # cf https://stackoverflow.com/questions/295135/turn-a-string-into-a-valid-filename
-        name = str(base64.urlsafe_b64encode(bytes(url, 'utf8')), encoding='utf8')
+        # replace everything except hyphen, word characters, underscore and dot
+        # with and underscore.
+        name = re.sub(r'(?u)[^-\w.]', '_', url)
```

Comment:
python-slugify seems to produce better results, if you don't mind an extra dependancy for this unimportant thing.